plugin/storage/integration/elasticsearch_test.go

```diff
@@ -184,6 +184,7 @@ func (s *ESStorageIntegration) initSpanstore(t *testing.T, allTagsAsFields, arch
 	require.NoError(t, err)
 	clientFn := func() estemplate.Client { return client }
 
+	// Initializing Span Reader and Writer
```

Comment:
thinking further through this, it's weird why we init the reader/writer manually instead of using es.Factory. But it's for another round of PRs.

Reply:
Yes exactly, I had the same doubt. Do you want me to initialize it using es.Factory instead?

Reply:
I'd say more important is to actually enable archive tests in other storages, that was the main point of this ticket, but right now it's all disabled except for ES.

Reply:
ah, good. There's no reason why Badger cannot support it, just point badger to another directory for data. Cassandra certainly supports it. Kafka isn't really a "storage" so yeah, it's ok to skip there.

Reply:
Yes, I didn't knew there was a workaround for badger just noticed it did not have CreateArchiveReader method and Stack Overflow said it didn't support archiving, so had skipped Badger. As for Cassandra, I was working on it but threw some error, so had removed the archive method, initially and would work on it after ES was fixed.

Reply:
What can I work on next, or should continue on this itself?

Reply:
up to you. May be worth finishing the ES by cleaning it up to make it use Factory rather than instantiate things manually.

Reply:
On it!